LinqGen.Generator/Instructions/Evaluations/Local/SingleEvaluation.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Cathei.LinqGen.Generator;
+
+public sealed class SingleEvaluation : LocalEvaluation {
+    private bool OrDefault { get; }
+
+    public SingleEvaluation(in LinqGenExpression expression, uint id, bool orDefault) : base(expression, id)
+    {
+        OrDefault = orDefault;
+    }
+
+    protected override TypeSyntax ReturnType => Upstream.OutputElementType;
+
+    protected override IEnumerable<StatementSyntax> RenderInitialization() {
+        if (Upstream.SupportCount) {
+            var expression = GreaterOrEqualExpression(Upstream.RenderCount()!, LiteralExpression(2));
+            if (!OrDefault)
+                expression = LogicalOrExpression(expression, BinaryExpression(SyntaxKind.EqualsExpression,
+                                                                              Upstream.RenderCount()!,
+                                                                              LiteralExpression(0)));
+            yield return IfStatement(expression, Block(SingletonList(ThrowInvalidOperationStatement())));
+        } else {
+            yield return LocalDeclarationStatement(LocalName("count").Identifier, LiteralExpression(0));
+            yield return LocalDeclarationStatement(ReturnType, LocalName("result").Identifier, DefaultLiteral);
+        }
+    }
+
+    protected override IEnumerable<StatementSyntax> RenderAccumulation() {
+        if (Upstream.SupportCount) {
+            yield return ReturnStatement(CurrentPlaceholder);
+        } else {
+            yield return IfStatement(GreaterOrEqualExpression(PostfixUnaryExpression(SyntaxKind.PostIncrementExpression, LocalName("count")), LiteralExpression(2)), Block(SingletonList(ThrowInvalidOperationStatement())));
```

Comment:
```suggestion
            yield return IfStatement(GreaterOrEqualExpression(PreIncrementExpression(LocalName("count")), LiteralExpression(2)), Block(SingletonList(ThrowInvalidOperationStatement())));
```

Seems like the exception only happens when element is more than 2, since currently it generates code like
```cs
if (count_9K9y03++ >= 2)
{
    ExceptionUtils.ThrowInvalidOperation();
}
```
So this does not throw when there is exactly 2 elements exist. Changing to prefix increment would solve it.